File: WerewolfParty-Server/DTO/PlayerRoleActionDto.cs
using WerewolfParty_Server.Enum;

namespace WerewolfParty_Server.DTO;

public class PlayerRoleActionDto 
{
    public int Id { get; set; }
    public required RoleName Role { get; set; }
    public required List<RoleActionDto> Actions { get; set; }
    
    public required bool isAlive { get; set; }
    public string Nickname { get; set; }
    public int AvatarIndex { get; set; }
}

File: apps/web/src/dto/PlayerRoleActionDto.ts
import { Role } from "@/enum/Role";
import { RoleActionDto } from "./RoleActionDto";

export interface PlayerRoleActionDto {
  id: number;
  nickname: string;
  avatarIndex: number;
  role: Role;
  actions: RoleActionDto[];
  isAlive: boolean;
}
